src/Units/KaijuActions.java

```diff
@@ -0,0 +1,943 @@
+import Engine.GameEvents.GameEventQueue;
+import Engine.GameEvents.HealUnitEvent;
+import Engine.GameEvents.MapChangeEvent;
+import Engine.GameEvents.MassDamageEvent;
+import Engine.GamePath.PathNode;
+import Engine.StateTrackers.StateTracker;
+import Engine.UnitActionLifecycles.TransformLifecycle;
+import Terrain.Environment;
+import Terrain.GameMap;
+import Terrain.MapLocation;
+import Terrain.MapMaster;
+import Terrain.TerrainType;
+import Units.KaijuWarsKaiju.HellTurkey;
+import Units.KaijuWarsKaiju.HellTurkeyLand;
+import Units.KaijuWarsKaiju.KaijuStateTracker;
+import Units.KaijuWarsKaiju.KaijuUnitModel;
+import Units.KaijuWarsUnits.KaijuWarsUnitModel;
+import Units.KaijuWarsUnits.Radar;
+
+public class KaijuActions
+{
+  public static class KaijuCrushFactory extends UnitActionFactory
+  {
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public GameActionSet getPossibleActions(GameMap map, GamePath movePath, Unit actor, boolean ignoreResident)
+    {
+      return new GameActionSet(new KaijuCrushAction(actor, movePath, this), false);
+    }
+
+    @Override
+    public String name(Unit actor)
+    {
+      return "CRUSH";
+    }
+  } //~Factory
+
+  /** List of things Kaiju break that aren't capturable */
+  public static final List<TerrainType> INERT_CRUSHABLES = Arrays.asList(TerrainType.BRIDGE, TerrainType.BUNKER, TerrainType.PILLAR, TerrainType.METEOR);
+  private static boolean isCrushable(MapLocation location)
+  {
+    return location.isCaptureable()
+        || INERT_CRUSHABLES.contains(location.getEnvironment().terrainType);
+  }
+
+  public static class KaijuCrushAction extends GameAction
+  {
+    private final KaijuCrushFactory type;
+    private final GamePath movePath;
+    private final XYCoord waitLoc;
+    private final Unit actor;
+
+    public KaijuCrushAction(Unit unit, GamePath path, KaijuCrushFactory pType)
+    {
+      type = pType;
+      actor = unit;
+      movePath = path;
+      if( (null != path) && (path.getPathLength() > 0) )
+      {
+        // Store the destination for later.
+        waitLoc = movePath.getEndCoord();
+      }
+      else
+        waitLoc = null;
+    }
+
+    @Override
+    public GameEventQueue getEvents(MapMaster gameMap)
+    {
+      // CRUSH actions consist of
+      //   sequence:
+      //     [Terrain destruction]
+      //      - [Heal for Big Donk]
+      //     [Mass damage]
+      //      - [Transform for Hell Turkey]
+      //     [Death]
+      //   MOVE
+      GameEventQueue crushEvents = new GameEventQueue();
+
+      // Validate input.
+      boolean isValid = true;
+      isValid &= null != actor && !actor.isTurnOver;
+      isValid &= (null != movePath) && (movePath.getPathLength() > 0);
+      isValid &= (null != gameMap);
+      isValid &= (actor.model instanceof KaijuUnitModel);
+
+      // Generate events.
+      if( isValid )
+      {
+        Utils.enqueueMoveEvent(gameMap, actor, movePath, crushEvents);
```

Comment:
The comment above says that the MOVE comes last, but it looks like this enqueues a full-length move up front before enqueuing the individual CRUSHes.

Reply:
The uh... the next line deletes it. I think that just got left in there when I did the iterative-move thing.

Reply:
So it does. Sounds like two lines we don't need anymore then.

Reply:
Something seemed off to me about that, and I looked again - `enqueueMoveEvent()` actually has a side-effect in that it snips the path if it will run into something.
OTOH, Kaiju are freely able to run through units, so I'm not sure it's needed?

I dunno. New code that breaks all the rules is hard to fit into your framework of the rules that it breaks.